src/android/com/adobe/phonegap/push/PushInstanceIDListenerService.java

```diff
@@ -5,23 +5,24 @@
 import android.content.SharedPreferences;
 import android.util.Log;
 
-import com.google.android.gms.iid.InstanceID;
-import com.google.android.gms.iid.InstanceIDListenerService;
+import com.google.firebase.iid.FirebaseInstanceId;
+import com.google.firebase.iid.FirebaseInstanceIdService;
 
 import org.json.JSONException;
 
 import java.io.IOException;
 
-public class PushInstanceIDListenerService extends InstanceIDListenerService implements PushConstants {
-    public static final String LOG_TAG = "PushPlugin_PushInstanceIDListenerService";
+public class PushInstanceIDListenerService extends FirebaseInstanceIdService implements PushConstants {
+    public static final String LOG_TAG = "PushPlugin_InsIdService";
 
     @Override
     public void onTokenRefresh() {
-        SharedPreferences sharedPref = getApplicationContext().getSharedPreferences(COM_ADOBE_PHONEGAP_PUSH, Context.MODE_PRIVATE);
-        String senderID = sharedPref.getString(SENDER_ID, "");
-        if (!"".equals(senderID)) {
-            Intent intent = new Intent(this, RegistrationIntentService.class);
-                startService(intent);
-        }
+        // Get updated InstanceID token.
+        String refreshedToken = FirebaseInstanceId.getInstance().getToken();
+        Log.d(LOG_TAG, "Refreshed token: " + refreshedToken);
+        // TODO: Implement this method to send any registration to your app's servers.
```

Comment:
Here we will need to add:

```
 // save new token
SharedPreferences.Editor editor = sharedPreferences.edit();
editor.putString(REGISTRATION_ID, token);
editor.commit();
```

to save the new token or the PushPlugin.java code may try to load the old token. Also, we'll need to figure out a way to fire a `registration` event.